app/models/course/assessment.rb

```diff
@@ -91,6 +91,16 @@ class Course::Assessment < ApplicationRecord
       )
   end)
 
+  # @!attribute [r] question_count
+  #   Gets the number of questions in this assessment.
+  #   @return [Integer]
+  calculated :question_count, (lambda do
+    Course::QuestionAssessment.unscope(:order).
+      select('coalesce(count(DISTINCT cqa.question_id), 0)').
+      joins('INNER JOIN course_question_assessments cqa ON cqa.assessment_id = course_assessments.id').
+      group('course_assessments.id')
```

Comment:
I just learned that since this method `calculated` is inside the model `Assessment`, the grouping is inherent here (meaning it will calculate for each `course_assessments.id`), and hence it won't be needed